dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -216,6 +246,29 @@ public BinaryRecordReader(InputStream is, long streamPos) throws Exception
         }
     }
 
+    private void startNewRecordBatchSpan()
+    {
+        if (this.recordBuilderSpan != null)
+        {
+            if (this.recordBatchSpan != null)
+            {
+                this.recordBatchSpan.setAttribute("RecordCount", (this.recordCount - this.recordBatchStart));
+                this.recordBatchSpan.setAttribute("DeserializationTimeMS", (this.recordDeserialTimeNS / 1000.0f));
+                this.recordBatchSpan.setAttribute("ExternalProcessingTimeMS", (this.externalProcessingTimeNS / 1000.0f));
+                this.recordBatchSpan.updateName( "RecordBatch[" + this.recordBatchStart + "," + this.recordCount + "]");
```

Comment:
are there other paths (exceptions) that would end the span under the original name?

Reply:
Good question, I added a check to close out the span when the input stream is closed and or when we are out of records.